### ISSUE TITLE
Public dashboards with private questions

### ISSUE THREAD

**Issue by orenmazor** [metabase/metabase#13172]
**Describe the bug**
If you create several private questions and an "our analytics" dashboard that uses them, the visiting user (not the creator) will see the following unhelpful error:

![Screen Shot 2020-08-26 at 2 43 04 PM](https://user-images.githubusercontent.com/384279/91343374-757ffc80-e7aa-11ea-9a89-64f55f95da73.png)

It took my users some time to hunt this one down, as most of the questions in the dashboard WERE in the shared space and just one was stopping the load of the dashboard.

**Logs**

there were some js console errors but I believe those are red herrings for this issue, which is simply an ACL error :)

**To Reproduce**
1. create several questions
2. move most questions into the shared space but keep some in the private space
3. create a shared dashboard using ALL questions (including the private one)
4. visit the dashboard from a new user

**Expected behavior**
one of a few things:

1. creating a shared dashboard automatically moves the component questions into the shared space (not so great)
2. a notification of some kind about private questions being used (ideal)
3. a 'permission denied' error when the new user loads the dashboard, indicating SOMETHING is not shared (probably easiest)

**Screenshots**

**Information about your Metabase Installation:**
```

  "browser-info": {
    "language": "en-CA",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:79.0) Gecko/20100101 Firefox/79.0",
    "vendor": ""
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.7+10",
    "java.vendor": "AdoptOpenJDK",
    "java.vendor.url": "https://adoptopenjdk.net/",
    "java.version": "11.0.7",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.7+10",
    "os.name": "Linux",
    "os.version": "4.14.181-108.257.amzn1.x86_64",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "postgres",
      "athena"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "11.8"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.2.8"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2020-07-21",
      "tag": "v0.36.0",
      "branch": "release-0.36.x",
      "hash": "b31049f"
    },
    "settings": {
      "report-timezone": null
    }
  }
}```

**Severity**
not blocking me, but might be annoying for users as they hit these repeatedly

**Additional context**
🤘

**Comment by flamber** [comment]
Hi @orenmazor 
This it was fixed in 0.36.1 - latest release is 0.36.4, so please try with that version.

**Comment by orenmazor** [comment]
thanks @flamber!
